### PR TITLE
Create FTS table + tests

### LINKED ISSUE #24: Implement full-text search in database
Using [FTS4](https://www.sqlite.org/fts3.html), implement full-text search in the database. FTS requires a virtual table that should mirror the existing "Features" table created in an earlier task. Any item that is inserted into the main features table should also be inserted into the virtual table.

Not all values from the original table need to be mirrored. For instance, the `json` values from the original table are basically useless for FTS.

Create a "search" query that returns all rows that match a given query string, as well as another query to clear the virtual table.

# Testing
Test the FTS implementation on the JVM.

1. Test that inserting a row into the original features table also inserts it into the virtual table.
2. Test that clearing the virtual table works
3. Insert a few values into the virtual table where some of the fields contain some text and test that the "search" query returns rows that match any or all of the text in those fields. Example: If a feature has a `common_name` of "River Building", the "search" query with "river" as an argument should return that row.

# Future Considerations

We may choose to use FTS5 in the future, but this will require more development work for a later time.

### PR DESCRIPTION
**Pull request by safa-howaid**
Resolves #24

### CODE REVIEW

File: common/src/androidTest/kotlin/DatabaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import com.cuhacking.atlas.db.Database.Companion.Schema
+import com.squareup.sqldelight.db.SqlDriver
+import com.squareup.sqldelight.sqlite.driver.JdbcSqliteDriver
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import kotlin.test.assertTrue
+
+lateinit var driver: SqlDriver
+lateinit var queries: FeatureQueries
```
Comment (dellisd) [inline common/src/androidTest/kotlin/DatabaseTest.kt:11]: These can be moved into the `DatabaseTest` class. The `queries` variable can be removed. We can/should just get the queries from the database at any time.

File: common/src/androidTest/kotlin/DatabaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        driver = JdbcSqliteDriver(JdbcSqliteDriver.IN_MEMORY)
+        Schema.create(driver)
+        val database = driver.createDatabase()
+        queries = database.featureQueries
+
+        val feature1 = Feature(1, "River Building","Richcraft Hall","Building",null,null,null,sampleFeature)
+        val feature2 = Feature(2, "RB 2311", null, "room","River Building", "RB2",null,sampleFeature)
+
+        queries.insertFeature(feature1)
+        queries.insertFeature(feature2)
```
Comment (dellisd) [inline common/src/androidTest/kotlin/DatabaseTest.kt:28]:
```suggestion
        database.featureQueries.run {
            insertFeature(feature1)
            insertFeature(feature2)
        }
```

We can use a [scope function](https://kotlinlang.org/docs/reference/scope-functions.html) here.
I'm not sure but we may even be able to add a query that allows multiple features to be inserted at once?

File: common/src/androidTest/kotlin/DatabaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    private val sampleFeature = GeoJsonAdapter.decode("""{"type":"Feature","geometry":{"type":"Point","coordinates":[-75.0,45.0]},"properties":{}}""")
+
+    @Before
+    fun setup() {
+        driver = JdbcSqliteDriver(JdbcSqliteDriver.IN_MEMORY)
+        Schema.create(driver)
+        val database = driver.createDatabase()
+        queries = database.featureQueries
+
+        val feature1 = Feature(1, "River Building","Richcraft Hall","Building",null,null,null,sampleFeature)
```
Comment (dellisd) [inline common/src/androidTest/kotlin/DatabaseTest.kt:24]: My advice would be to take these (and any other) features and declare them outside of this function.
Then, add a helper extension function that converts a `Feature` into its equivalent `Feature_fts`. 

This way you have a set of features that you can use anywhere, and then the helper function will make it more reusable.

File: common/src/androidTest/kotlin/DatabaseTest.kt (hunk 1 above)
Comment (safa-howaid) [reply common/src/androidTest/kotlin/DatabaseTest.kt:11]: I believe there isn't a query that allows the insertion of multiple queries at once, so a scope function would be the best solution like you mentioned!

File: common/src/androidTest/kotlin/DatabaseTest.kt (hunk 2 above)
Comment (safa-howaid) [reply common/src/androidTest/kotlin/DatabaseTest.kt:28]: I believe there isn't a query that allows the insertion of multiple queries at once, so a scope function would be the best solution like you mentioned!

File: common/src/androidTest/kotlin/DatabaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    fun close(){
+        driver.close()
+    }
+
+    @Test
+    fun `check that inserting into feature also inserts into feature_fts`(){
+        database.featureQueries.insertFeature(feature3)
+
+        assertTrue(database.featureQueries.search("HS 1301B").executeAsList()
+                .contains(featureToFTS(feature3))
```
Comment (dellisd) [inline common/src/androidTest/kotlin/DatabaseTest.kt:44]:
```suggestion
                .contains(feature3.toFTS())
```

Usage of extension function

File: common/src/androidTest/kotlin/DatabaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        )
+    }
+
+    @Test fun `check that search query works`(){
+        assertTrue(database.featureQueries.search("river").executeAsList().containsAll(
+                listOf(featureToFTS(feature1), featureToFTS(feature2))
```
Comment (dellisd) [inline common/src/androidTest/kotlin/DatabaseTest.kt:50]:
```suggestion
                listOf(feature1.toFTS(), feature2.toFTS())
```

**Pull request merged**